src/cli.rs

```diff
@@ -144,6 +144,32 @@ impl From<OutputFormat> for OutputFormatter {
     }
 }
 
+async fn read_popgetter(config: Config) -> anyhow::Result<Popgetter> {
+    let xdg_dirs = xdg::BaseDirectories::with_prefix("popgetter")?;
+    // TODO: enable cache to be optional
+    let path = xdg_dirs.get_cache_home();
+    // Try to read metadata from cache
+    if path.exists() {
+        match Popgetter::new_with_config_and_cache(config.clone(), &path) {
+            Ok(popgetter) => return Ok(popgetter),
+            Err(err) => {
+                error!("Failed to read metadata from cache with error: {err}");
```

Comment:
Add comment explaining why error not propagated
```suggestion
                // Log error, continue without cache and attempt to create one
                error!("Failed to read metadata from cache with error: {err}");
```